backend/src/apiserver/server/api_converter_test.go

```diff
@@ -2306,3 +2314,2356 @@ func TestToApiRecurringRun(t *testing.T) {
+			model.RuntimeStatePending,
+			false,
+			"",
+		},
+		{
+			"V1 Running",
+			"Running",
+			model.RuntimeStateRunningV1,
+			model.RuntimeStateRunning,
+			false,
+			"",
+		},
+		{
+			"V1 Succeeded",
+			"Succeeded",
+			model.RuntimeStateSucceededV1,
+			model.RuntimeStateSucceeded,
+			false,
+			"",
+		},
+		{
+			"V1 Skipped",
+			"Skipped",
+			model.RuntimeStateSkippedV1,
+			model.RuntimeStateSkipped,
+			false,
+			"",
+		},
+		{
+			"V1 Failed",
+			"Failed",
+			model.RuntimeStateFailedV1,
+			model.RuntimeStateFailed,
+			false,
+			"",
+		},
+		{
+			"V1 Error",
+			"Error",
+			model.RuntimeStateFailedV1,
+			model.RuntimeStateFailed,
+			false,
+			"",
+		},
+		{
+			"V1 Empty",
+			"",
+			model.RuntimeStateUnknownV1,
+			model.RuntimeStateUnspecified,
+			false,
+			"",
+		},
+		{
+			"V1 Unknown",
+			"Unknown",
+			model.RuntimeStateUnknownV1,
+			model.RuntimeStateUnspecified,
+			false,
+			"",
+		},
+		{
+			"V1 NO_STATUS",
+			"NO_STATUS",
+			model.RuntimeStateUnknownV1,
+			model.RuntimeStateUnspecified,
+			false,
+			"",
+		},
+		{
+			"V1 Terminating",
+			"Terminating",
+			model.RuntimeStateTerminatingV1,
+			model.RuntimeStateCancelling,
+			false,
+			"",
+		},
+		{
+			"V1 Ready",
+			"Ready",
+			model.RuntimeStateRunningV1,
+			model.RuntimeStateRunning,
+			false,
+			"",
+		},
+		{
+			"V1 Done",
+			"Done",
+			model.RuntimeStateSucceededV1,
+			model.RuntimeStateSucceeded,
+			false,
+			"",
+		},
+		{
+			"V1 wrong value",
+			"wrong value",
+			model.RuntimeStateUnknownV1,
+			model.RuntimeStateUnspecified,
+			false,
+			"",
+		},
+
+		{
+			"V2 RUNTIME_STATE_UNSPECIFIED",
+			"RUNTIME_STATE_UNSPECIFIED",
+			model.RuntimeStateUnknownV1,
+			model.RuntimeStateUnspecified,
+			false,
+			"",
+		},
+		{
+			"V2 RUNNING",
+			"RUNNING",
+			model.RuntimeStateRunningV1,
+			model.RuntimeStateRunning,
+			false,
+			"",
+		},
+		{
+			"V2 SUCCEEDED",
+			"SUCCEEDED",
+			model.RuntimeStateSucceededV1,
+			model.RuntimeStateSucceeded,
+			false,
+			"",
+		},
+		{
+			"V2 SKIPPED",
+			"SKIPPED",
+			model.RuntimeStateSkippedV1,
+			model.RuntimeStateSkipped,
+			false,
+			"",
+		},
+		{
+			"V2 CANCELED",
+			"CANCELED",
+			model.RuntimeStateFailedV1,
+			model.RuntimeStateCanceled,
+			false,
+			"",
+		},
+		{
+			"V2 PAUSED",
+			"PAUSED",
+			model.RuntimeStatePendingV1,
+			model.RuntimeStatePaused,
+			false,
+			"",
+		},
+		{
+			"V2 Empty",
+			"",
+			model.RuntimeStateUnknownV1,
+			model.RuntimeStateUnspecified,
+			false,
+			"",
+		},
+		{
+			"V2 PENDING",
+			"PENDING",
+			model.RuntimeStatePendingV1,
+			model.RuntimeStatePending,
+			false,
+			"",
+		},
+		{
+			"V2 RuntimeState_CANCELED",
+			apiv2beta1.RuntimeState_CANCELED,
+			model.RuntimeStateFailedV1,
+			model.RuntimeStateCanceled,
+			false,
+			"",
+		},
+		{
+			"nil",
+			nil,
+			model.RuntimeStateUnknownV1,
+			model.RuntimeStateUnspecified,
+			false,
+			"",
+		},
+		{
+			"V2 int32=7",
+			int32(7),
```

Comment:
Is this a possible value we could get? Is 7 the enum value?
And we expect a filed state for v1 yet canceled state for v2?

Reply:
You are right. These test cases are completely unneccessary. I removed them and simplified the function too.